src/log_message.ts

```diff
@@ -37,19 +38,108 @@ type LogMessage = {
   Source: string;
 };
 
-export function parse(buff: ArrayBuffer): (MetadataObject | false)[] {
-  return inflate(buff, { to: "string" })
-    .trim()
-    .split("\n")
-    .map((line) => {
-      try {
-        // We are only returning the matched data contents
-        // because that is only what we care about in this script
-        // but the entire message could be returned if desired
-        const { Metadata }: LogMessage = JSON.parse(line);
-        return Metadata ?? false;
-      } catch {
-        return false;
+export function parse(
+  rs: ReadableStream,
+  MATCHED_PAYLOAD_PRIVATE_KEY: string
+): ReadableStream {
+  let temp = "";
+  let firstItem = true;
+
+  const textTransform = new TransformStream<string, string>({
+    start(controller) {
+      controller.enqueue("[");
+    },
+    async transform(chunk, controller) {
+      if (!chunk) return;
+      const { returnMsgs, buffer } = await updateStream(
+        temp + chunk,
+        MATCHED_PAYLOAD_PRIVATE_KEY
+      );
+      temp = buffer;
+      if (returnMsgs.length > 0) {
+        const json = JSON.stringify(returnMsgs).slice(1, -1);
+        controller.enqueue(firstItem ? json : "," + json);
+        firstItem = false;
+      }
+    },
+    async flush(controller) {
+      if (temp.length > 0) {
+        const { returnMsgs, buffer } = await updateStream(
+          temp,
+          MATCHED_PAYLOAD_PRIVATE_KEY,
+          true
+        );
+        if (buffer) {
+          console.error("Buffer not empty at flush:", buffer);
+        }
+        if (returnMsgs.length > 0) {
+          controller.enqueue("," + JSON.stringify(returnMsgs).slice(1, -1));
+        }
+      }
+      controller.enqueue("]");
+      controller.terminate();
+    },
+  });
+
+  const encodedStream = rs
+    .pipeThrough(new DecompressionStream("gzip"))
+    .pipeThrough(new TextDecoderStream("utf-8"))
+    .pipeThrough(textTransform)
+    .pipeThrough(new TextEncoderStream());
+
+  return encodedStream;
+}
+
+export async function updateStream(
+  txt: string,
+  MATCHED_PAYLOAD_PRIVATE_KEY: string,
+  lastBatch: boolean = false
+) {
+  let buffer = "";
+  let returnMsgs: LogMessage[] = [];
+  if (txt.includes("\n")) {
```

Comment:
this logic needs a lot of cleanup, there sees to be a lot of copy/paste going on here.